twingly_search/errors.py

```diff
@@ -1,21 +1,64 @@
-class TwinglyException(Exception):
-    def from_api_response_message(self, message):
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+
+class Error(object):
+    """
+    Error response from the Search API
+    
+    Attributes:
+        code    (string) error code
+        message (string) error message
+    """
+
+    def __init__(self, code='', message=''):
+        self.code = code
+        self.message = message
+
+    def __unicode__(self):
+        return "Error code: %s. Error message: %s." % (self.code, self.message)
+
+    def __str__(self):
+        return self.__unicode__()
+
+
+class TwinglySearchException(Exception):
+    @classmethod
+    def from_api_response_message(cls, error):
         """
-        :param message: API response error message
+        Create TwinglySearchException from the Error
+        :param error: error from the API response
         """
-        if 'API key' in message:
-            raise TwinglyAuthException(message)
-        else:
-            raise TwinglyServerException(message)
+        if error is not None and error.code is not None:
+            if error.code.startswith('4'):
+                if error.code.startswith('400') or error.code.startswith('404'):
+                    raise TwinglySearchQueryException(error)
+                if error.code.startswith('401') or error.code.startswith('402'):
+                    raise TwinglySearchAuthenticationException(error)
+                raise TwinglySearchClientException(error)
+            if error.code.startswith('5'):
+                raise TwinglySearchServerException(error)
+            raise TwinglySearchErrorException(error)
+
+
+class TwinglySearchErrorException(TwinglySearchException):
```

Comment:
Why was this class added? Wouldn't it work with just having `TwinglySearchException` as the base class for the other exceptions?

I also noticed that the exception class hierarchy has changed a bit below. Any particular reason?